File: examples/python/models/Square.py
from .ShapeInterface import ShapeInterface
from .ManageShapeInterface import ManageShapeInterface

class Square(ShapeInterface, ManageShapeInterface):
    def __init__(self, length: int):
        self.length = length

    def area(self) -> float:
        return self.length ** 2

    def calculate(self) -> float:
        return self.area()

File: examples/csharp/Model/Square.cs
namespace Model
{
    public class Square: IShape, IManageShape
    {
        public int Length { get; private set; }

        public Square(int length)
        {
            this.Length = length;
        }

        public double Area()
        {
            return this.Length * this.Length;
        }

        public double Calculate()
        {
            return this.Area();
        }
    }
}
